vms/platformvm/txs/mempool/mempool.go

```diff
@@ -73,6 +67,12 @@ type Mempool interface {
 	// It's guaranteed that the returned tx, if not nil, is a StakerTx.
 	PeekStakerTx() *txs.Tx
 
+	// RequestBuildBlock notifies the consensus engine that a block should be
+	// built. If [emptyBlockPermitted] is true, the notification will be sent
+	// if there are no transactions in the mempool. Otherwise, a notification
+	// will only be sent if there is at least one transaction in the mempool.
+	RequestBuildBlock(emptyBlockPermitted bool)
```

Comment:
I'd like to get rid of the boolean parameter here. `RequestBuildBlock` is called with false only if a tx has just been added to the mempool; moreover every time we add a transaction to the mempool, we then request the engine to build the block.
So I'd suggest we:

1. Turn` Add(*txs.Tx)` into `Add([]*txs.Tx)`
2. Ping the engine directly at the end of Add(, instead of asking the Add client to do it
3. Drop the boolead

Reply:
Having the engine be pinged directly at the end of `Add` is a bit strange since we'll have two ways to request a block to be built (`Add` or `RequestBuildBlock`). I'd prefer to be consistent and have the mempool client always be responsible for calling `RequestBuildBlock` instead of just in the case there are no txs

The boolean override isn't great but I think it's better to be explicit: "I want to build a block and am OK with it being empty"

Reply:
We already have two ways to build a block rn, from mempool txs and from stakers events. No added strangeness.
Every time a client adds a transaction to the mempool. it has to ping the engine to build a block out of it. I think we don't need the client to do the work here, we can automate this action by letting the mempool it itself as soon as it has added the tx. We would remove some client code too.
I don't think it really matters if the block is empty or not. For sure if we automatize the mempool pinging the engine we can have a simpler interface (no boolean) and we only have the block builder really in need to requesto block building outside the mempool.

Reply:
I coded my suggestion here https://github.com/ava-labs/avalanchego/pull/2346